homeassistant/components/compensation/sensor.py

```diff
@@ -45,43 +44,42 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     )
 
 
-class CompensationSensor(Entity):
+class CompensationSensor(SensorEntity):
     """Representation of a Compensation sensor."""
 
     def __init__(
         self,
-        hass,
-        entity_id,
-        name,
+        source,
+        unique_id,
         attribute,
         precision,
         polynomial,
         unit_of_measurement,
     ):
         """Initialize the Compensation sensor."""
-        self._entity_id = entity_id
-        self._name = name
+        self._source_entity_id = source
         self._precision = precision
-        self._attribute = attribute
+        self._source_attribute = attribute
         self._unit_of_measurement = unit_of_measurement
         self._poly = polynomial
         self._coefficients = polynomial.coefficients.tolist()
-        self._state = STATE_UNKNOWN
+        self._state = None
+        self._unique_id = unique_id
 
     async def async_added_to_hass(self):
         """Handle added to Hass."""
         self.async_on_remove(
             async_track_state_change_event(
                 self.hass,
-                [self._entity_id],
+                [self._source_entity_id],
                 self._async_compensation_sensor_state_listener,
             )
         )
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
```

Comment:
We still want to set a default name that is better than the common default name.